_posts/2019-02-12-native-elm-datepicker.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: post
+title: "Native Elm Datepicker"
+date: 2019-04-12 5:00:00 -0500
+category: blog
+author: RJ
+author_twitter:
+image:
+---
+
+When browser support for date input types was worse (or non-existent), JavaScript datepickers were a common solution if you needed to allow a user to enter a date on a form. These days, browser support is much better, making native browser datepickers a much more viable option. If you need a datepicker in Elm, there is a package that will meet your needs ([CurrySoftware/elm-datepicker](https://github.com/CurrySoftware/elm-datepicker)), but it is more cumbersome than the native browser option, which in Elm is pretty easy to use.
+
+To use the native browser datepicker in Elm all you need to do is add an input tag, capture its input events with `Html.Events.onInput`, and parse the value of the input event for a date. The value will be a string containing a date in the ISO 8601 format (YYYY-MM-DD).
+
+Here is very simple Ellie example demonstrating how it works:
+
+<iframe src="https://ellie-app.com/embed/3RjFHN9sHzca1" style="width:100%; height:400px; border:0; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
```

Comment:
I hate iframes, but for the purpose of embedding Ellie, I think that is very cool.

Reply:
Tis the only way that Ellie offers